superliminal.asl

```diff
@@ -273,6 +327,22 @@ split
 
             const string Retrospect = "Assets/_Levels/_LiveFolder/ACT03/EndingMontage/EndingMontage_Live.unity";
             finalAlarmClicked = current.scene == Retrospect && current.alarmStopped;
+            
+            bool diffFireAlarm = !arrayEquals(current.statusFireAlarm, old.statusFireAlarm);
```

Comment:
Use `Enumerable.SequenceEqual` here, and check for `settings["split_on_FireAlarm"]` **first** to avoid comparing the arrays when not needed.
```suggestion
            bool diffFireAlarm = settings["split_on_FireAlarm"] && !Enumerable.SequenceEqual(current.statusFireAlarm, old.statusFireAlarm);
```
etc.